app/models/video.rb

```diff
@@ -5,4 +5,10 @@ class Video < ApplicationRecord
   has_many :users, through: :user_videos
   belongs_to :tutorial
   validates_presence_of :position
+
+  def self.ordered_grouped_videos
+    self.select("videos.tutorial_id, videos.title, videos.position")
+    .order(:position)
+    .group_by(&:tutorial_id)
+  end
```

Comment:
We solved this problem, although unconventionally! We can take a look at it again later if we have time, but I am glad that we got it working